### ISSUE TITLE
Incorrect treatment of columns that include space in their names

### ISSUE THREAD

**Issue by TimurMedzhitov** [pawelsalawa/sqlitestudio#4109]
### Details
If the query uses space(s) in column names, no errors are raised and in some cases even incorrect result is returned.
The cases I have explored was when a column with spaces like [GEO LOC XXX] was used in WHERE, HAVING and aggregate function. Similar behaviour most likely exists in other cases when incorrect column name with spaces is used.

### Steps to reproduce
CREATE TABLE IF NOT EXISTS T0001(
ID INTEGER PRIMARY KEY,
[GEO LOC] TEXT
);

INSERT INTO T0001([GEO LOC]) VALUES('Ontario');
INSERT INTO T0001([GEO LOC]) VALUES('Alberta');

-- AS EXPECTED, RESULT IS RETURNED
SELECT * FROM T0001
WHERE [GEO LOC] = 'Ontario';

-- AS EXPECTED, ERROR IS RAISED
SELECT * FROM T0001
WHERE [GEOLOC] = 'Ontario';

-- AS EXPECTED, ERROR IS RAISED
SELECT * FROM T0001
WHERE GEOLOC = 'Ontario';

-- NOT AS EXPECTED, SHOULD RAISE ERROR NO SUCH COLUMN 'GEO LOC XXX'
SELECT * FROM T0001;
WHERE [GEO LOC XXX] = 'Ontario';

-- AS EXPECTED, RESULT IS RETURNED
SELECT ID, COUNT([GEO LOC]) FROM T0001
GROUP BY [ID]
HAVING [ID] >= 1;

-- NOT AS EXPECTED, SHOULD RAISE ERROR NO SUCH COLUMN 'ID XXX'
SELECT ID, COUNT([GEO LOC]) FROM T0001
GROUP BY [ID XXX]
HAVING [ID XXX] >= 1;

-- NOT AS EXPECTED, SHOULD RAISE ERROR NO SUCH COLUMN 'GEO LOC XXX'
SELECT ID, COUNT([GEO LOC XXX]) FROM T0001
GROUP BY ID
HAVING ID >= 1;

### Operating system
macOS Catalina 10.15.7

### SQLiteStudio version
SQLiteStudio 3.3.3

**Comment by pawelsalawa** [comment]
Yes, there is a problem with using incorrect column names that require quoting (the `[ ]`). It's a bug to be fixed.

What SQLiteStudio is doing it's parsing the query, applying all necessary metadata parts and then re-assemble the query, but during re-assembling SQLiteStudio picks `" "` for quoting instead of `[ ]` and it just happens that `" "` is interpreted by SQLite as either quoted object name, or as a string. Since the this object here is not valid name of column, the fallback strategy is to treat it as a string, which results in always negative condition result.
